solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -716,55 +717,85 @@ contract RandomBeacon is Ownable {
     }
 
     /// @notice Creates a new relay entry.
+    /// @dev This function can be called only before the soft timeout. This is a
+    ///      happy scenario without slashing.
+    /// @param entry Group BLS signature over the previous entry.
+    function submitRelayEntry(bytes calldata entry) external {
+        require(relay.isRequestInProgress(), "No relay request in progress");
+        require(
+            !relay.hasRequestSoftTimedOut(),
+            "Relay submission passed a soft timeout"
+        );
+
+        Groups.Group storage group = groups.getGroup(
+            relay.currentRequestGroupID
+        );
+
+        relay.submitEntry(entry, group.groupPubKey);
+
+        // If DKG is awaiting a seed, that means the we should start the actual
+        // group creation process.
+        if (dkg.currentState() == DKG.State.AWAITING_SEED) {
+            dkg.start(uint256(keccak256(entry)));
+        }
+
+        callback.executeCallback(uint256(keccak256(entry)), callbackGasLimit);
+    }
+
+    /// @notice Creates a new relay entry.
+    /// @dev It should be called only after a soft timeout. Group members will be
+    ///      slashed. Function reverts if a hard timeout is hit.
     /// @param entry Group BLS signature over the previous entry.
     /// @param groupMembers Group member ids that participated in dkg (excluding IA/DQ).
     function submitRelayEntry(
         bytes calldata entry,
         uint32[] calldata groupMembers
     ) external {
-        uint256 currentRequestId = relay.currentRequestID;
+        require(relay.isRequestInProgress(), "No relay request in progress");
```

Comment:
1. We did, but it was inside of `hasRequestTimedOut`. I decoupled it to avoid duplication of checking `isRequestInProgress()` twice, cause `submitEntry()` called it first at the very beginning of the function and the second time inside of `hasRequestTimedOut`. Now we have two functions to check for timeouts: soft and hard which are used in `submitRelayEntry`. I thought it would add another duplication (3 in total) if leaving `isRequestInProgress` inside of timeout checks. Pls also see [this comment](https://github.com/keep-network/keep-core/pull/2772#discussion_r776761491) which adds a bit more context.

2. Yes, it was harder to test this particular check and that's why I moved it here but lemme see what I can do in tests to move it back where it was.

Reply:
As per 2) 6505ecd